### PR TITLE
skip tautological monomial constraints, closing #633

### PR DESCRIPTION
**Pull request by bqpd**
@whoburg, waiting on review.

### CODE REVIEW

File: gpkit/nomials/nomial_math.py [hunk 1]
```diff
@@ -473,8 +473,10 @@ def _simplify_posy_ineq(self, exps, cs):
             if not exps[0]:
                 if cs[0] > 1:
                     raise ValueError("infeasible constraint: %s" % self)
-            # for now, we allow tautological monomial constraints (cs[0] <= 1)
-            # because they allow models to impose requirements on variables
+                else:
+                    # allow tautological monomial constraints (cs[0] <= 1)
+                    # because they allow models to impose requirements
+                    return [], []
```
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: should it return `(), ()` to match the tuple return type below?

File: gpkit/nomials/nomial_math.py [hunk 2]
```diff
@@ -523,9 +525,11 @@ def as_posyslt1(self):
             # remove any cs that are just nans and/or 0s
             nans = np.isnan(cs)
             if np.all(nans) or np.all(cs[~nans] == 0):
-                return []  # skip nan'd or 0'd constraint
+                continue  # skip nan'd or 0'd constraint
```
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: was this a bug catch?

**Comment by whoburg** [comment]
looks great. See one inline comment; make a change if you agree, and merge either way.

File: gpkit/nomials/nomial_math.py (hunk 2 above)
Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: Yup.

File: gpkit/nomials/nomial_math.py (hunk 1 above)
Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: Hmm, it's really tuple/ndarray

File: gpkit/nomials/nomial_math.py (hunk 2 above)
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: :+1:

File: gpkit/nomials/nomial_math.py (hunk 1 above)
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: I guess so, if they're not expensive to create. I think array not ndarray though -- I for one don't know how to create an empty ndarray:

```python
In [1]: import numpy as np

In [2]: np.ndarray()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-98b15771a5de> in <module>()
----> 1 np.ndarray()

TypeError: Required argument 'shape' (pos 1) not found

In [3]: np.ndarray([])
Out[3]: array(9.223372036854776e+18)

In [4]: np.array([])
Out[4]: array([], dtype=float64)
```

**Comment by whoburg** [comment]
#633 is closed but this hasn't been merged yet -- should we merge it?

**Comment by bqpd** [comment]
Oh, derp. I think so!

**Comment by bqpd** [comment]
Responded to code review, ready to merge.

**Pull request merged**